x/stakeibc/keeper/delegation.go

```diff
@@ -31,3 +37,61 @@ func (k Keeper) RemoveDelegation(ctx sdk.Context) {
 	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefix(types.DelegationKey))
 	store.Delete([]byte{0})
 }
+
+// ProcessDelegationStaking goes through each HostZone and stakes the delegation
+func (k Keeper) ProcessDelegationStaking(ctx sdk.Context) {
+	icaStake := func(index int64, zoneInfo types.HostZone) (stop bool) {
+		// Verify the delegation ICA is registered
+		k.Logger(ctx).Info(fmt.Sprintf("\tProcessing delegation %s", zoneInfo.ChainId))
+		delegationIca := zoneInfo.GetDelegationAccount()
+		if delegationIca == nil || delegationIca.Address == "" {
+			k.Logger(ctx).Error("Zone %s is missing a delegation address!", zoneInfo.ChainId)
+			return false
+		}
+		cdc := k.cdc
+		DelegateOnHost := k.DelegateOnHost
+
+		var queryBalanceCB icqkeeper.Callback = func(k icqkeeper.Keeper, ctx sdk.Context, args []byte, query icqtypes.Query) error {
+			k.Logger(ctx).Info(fmt.Sprintf("\tdelegation staking callback on %s", zoneInfo.HostDenom))
```

Comment:
I believe you want to use the parent function's keeper. Easy fix it to rename `k icqkeeper.Keeper` on line 54 to `icqk`

Reply:
Great catch, done.